chartpress.py

```diff
@@ -1026,10 +1117,16 @@ def main(args=None):
     #   - push chart (--publish-chart, --extra-message)
     for chart in config["charts"]:
         forced_version = None
+        # TODO: maybe warn and switch default in the future?
```

Comment:
While we are not switching the default in this PR, should we plan for it?

- If chartpress is used to just publsih releases along with a git tag is pushed, then switching the default to `useChartVersion: true` is adding trouble.
- If using it to publish between git tags as well, then both options work, but I'm +1 for switching the default.

I understand it as switching the default would imply forcing simple use cases (jupyterhub/pebble-helm-chart, dask/dask-gateway, and dask/dask-kubernetes) would need to manually declare the Chart.yaml version or learn about this and opt out even though they don't need the functionality.

I'm inclined to think we shouldn't switch, but instead ask that those that wants to emit dev releases in this more nice way also take on the work of maintaining Chart.yaml's version themselves.

```suggestion
```

Reply:
Let's make it purely opt-in for now, test it out a few places (e.g. z2jh), and see how we feel. If it seems like a pure win, we can make a patch release notifying that we'll change the default, otherwise let's leave it alone.